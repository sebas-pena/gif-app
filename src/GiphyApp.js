import React, { useState } from 'react'
import { AddCategory } from './component/AddCategory'
import { GifGallery } from './component/GifGallery'


export const GiphyApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h1>AGAUGA</h1>
            <p><b>Another Gif App Using Giphy API</b></p>
            <AddCategory setCategories={setCategories} categories={categories} />
            <hr />

            <ol>
                {
                    categories.map(
                        category => <GifGallery key={category} category={category} />
                    )
                }
            </ol>

        </>
    )

}