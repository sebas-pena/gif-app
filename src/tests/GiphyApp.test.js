import React from 'react';
import { shallow } from 'enzyme'
import { GiphyApp } from '../GiphyApp';


describe('Pruebas sobre <GiphyApp />', () => {

    test('Debe renderizar <GiphyApp /> correctamente', () => {

        const wrapper = shallow(<GiphyApp />)
        expect(wrapper).toMatchSnapshot()

    })

    test('Debe renderizar la cantidad correctas de listas', () => {

        const categories = ['imagine dragons', 'AJR']
        const wrapper = shallow(<GiphyApp defaultCategories={categories} />)
        expect(wrapper.find('GifGallery').length).toBe(categories.length)

    })

})