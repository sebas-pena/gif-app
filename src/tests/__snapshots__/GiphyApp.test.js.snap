// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GiphyApp /> Debe renderizar <GiphyApp /> correctamente 1`] = `
<Fragment>
  <h1>
    AGAUGA
  </h1>
  <p>
    <b>
      Another Gif App Using Giphy API
    </b>
  </p>
  <AddCategory
    categories={Array []}
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
